### ISSUE TITLE
Add support for file-related subset of java.io.*

### ISSUE THREAD

**Issue by densh** [scala-native/scala-native#572]
This issue supersedes #311 and is based on linking errors we've seen in the wild so far:

```
@java.io.BufferedInputStream
@java.io.BufferedInputStream::init_class.java.io.InputStream
@java.io.BufferedInputStream::read_i32
@java.io.BufferedOutputStream
@java.io.BufferedOutputStream::init_class.java.io.OutputStream
@java.io.BufferedReader::lines_trait.java.util.stream.Stream
@java.io.BufferedWriter
@java.io.BufferedWriter::init_class.java.io.Writer
@java.io.Console
@java.io.DataOutputStream
@java.io.DataOutputStream::init_class.java.io.OutputStream
@java.io.DataOutputStream::write_class.ssnr.ByteArray_i32_i32_unit
@java.io.File$
@java.io.File$::field.pathSeparator
@java.io.File$::field.separator
@java.io.File$::field.separatorChar
@java.io.File::canExecute_bool
@java.io.File::canRead_bool
@java.io.File::canWrite_bool
@java.io.File::createNewFile_bool
@java.io.File::deleteOnExit_unit
@java.io.File::exists_bool
@java.io.File::getAbsoluteFile_class.java.io.File
@java.io.File::getAbsolutePath_class.java.lang.String
@java.io.File::getCanonicalPath_class.java.lang.String
@java.io.File::getName_class.java.lang.String
@java.io.File::getParent_class.java.lang.String
@java.io.File::getPath_class.java.lang.String
@java.io.File::hashCode_i32
@java.io.File::isAbsolute_bool
@java.io.File::isDirectory_bool
@java.io.File::isFile_bool
@java.io.File::lastModified_i64
@java.io.File::length_i64
@java.io.File::listFiles_class.ssnr.ObjectArray
@java.io.File::listFiles_trait.java.io.FileFilter_class.ssnr.ObjectArray
@java.io.File::mkdir_bool
@java.io.File::mkdirs_bool
@java.io.File::toURI_class.java.net.URI
@java.io.FileFilter
@java.io.FileFilter::accept_class.java.io.File_bool
@java.io.FileInputStream
@java.io.FileInputStream::close_unit
@java.io.FileInputStream::getChannel_class.java.nio.channels.FileChannel
@java.io.FileInputStream::init_class.java.io.File
@java.io.FileNotFoundException
@java.io.FileNotFoundException::init_class.java.lang.String
@java.io.FileOutputStream
@java.io.FileOutputStream::init_class.java.io.File
@java.io.FileOutputStream::init_class.java.io.File_bool
@java.io.FileOutputStream::init_class.java.lang.String
@java.io.FileReader
@java.io.FileReader::init_class.java.io.File
@java.io.FileWriter
@java.io.FileWriter::init_class.java.io.File_bool
@java.io.LineNumberReader
@java.io.LineNumberReader::getLineNumber_i32
@java.io.LineNumberReader::init_class.java.io.Reader
@java.io.PushbackReader
@java.io.PushbackReader::init_class.java.io.Reader
@java.io.PushbackReader::unread_i32_unit
@java.io.RandomAccessFile
@java.io.RandomAccessFile::getChannel_class.java.nio.channels.FileChannel
@java.io.RandomAccessFile::init_class.java.io.File_class.java.lang.String
```

Blocks: #557, #575 

/cc @Duhemm

**Comment by densh** [comment]
**Update:** just added linking errors from betterfiles.

**Comment by densh** [comment]
Fixed in #574
